File: modules/agent/src/models/Job.py
from sqlalchemy import Column, Text, DateTime, BigInteger, ForeignKey
from sqlalchemy.orm import relationship
from models import Base

"""Data models for jobs (extends Lead via Joined Table Inheritance).

SQLAlchemy model for database persistence (unavoidable OOP requirement).
Functional TypedDict models for business logic.
"""




# SQLAlchemy model (OOP required for ORM)
class Job(Base):
    """Job SQLAlchemy model (extends Lead via Joined Table Inheritance)."""

    __tablename__ = "Job"

    id = Column(BigInteger, ForeignKey("Lead.id", ondelete="CASCADE"), primary_key=True)
    job_title = Column(Text, nullable=False)
    job_url = Column(Text, nullable=True)
    description = Column(Text, nullable=True)
    resume_date = Column(DateTime(timezone=True), nullable=True)
    salary_range = Column(Text, nullable=True)  # Legacy field, kept for backwards compat
    salary_range_id = Column(BigInteger, ForeignKey("Salary_Range.id", ondelete="SET NULL"), nullable=True)

    # Relationships
    lead = relationship("Lead", back_populates="job")
    salary_range_ref = relationship("SalaryRange")

File: modules/agent/internal/models/Job.go
package models

import "time"

type Job struct {
	ID                   int64      `gorm:"primaryKey" json:"id"` // FK to Lead.id
	JobTitle             string     `gorm:"not null" json:"job_title"`
	Description          *string    `json:"description"` // was "notes", renamed in migration 043
	JobURL               *string    `json:"job_url"`
	ResumeDate           *time.Time `json:"resume_date"`
	DatePosted           *time.Time `json:"date_posted"`
	DatePostedConfidence *string    `json:"date_posted_confidence"` // high, medium, low, none
	SalaryRange          *string    `json:"salary_range"`           // Legacy field
	SalaryRangeID        *int64     `json:"salary_range_id"`

	// Relations
	Lead           *Lead        `gorm:"foreignKey:ID" json:"lead,omitempty"`
	SalaryRangeRef *SalaryRange `gorm:"foreignKey:SalaryRangeID" json:"salary_range_ref,omitempty"`
}

func (Job) TableName() string {
	return "Job"
}
